client/packages/core/src/Reactor.js

```diff
@@ -546,17 +546,11 @@ export default class Reactor {
       return;
     }
 
-    const q = msg.q || msg["original-event"]?.q;
-    if (q) {
+    const q = msg["original-event"]?.q;
+    if (q && msg["original-event"]?.op === "add-query") {
```

Comment:
I updated this to only handle the error if the original op was `add-query`. `remove-query` also sets a `q` field, so this will prevent us from getting ourselves into a loop where `remove-query` fails so we send another `remove-query` which fails, etc.

I also deleted the check for `msg.q` because we never set a top-level `q` field in the error handler.